src/io/io.ts

```diff
@@ -89,27 +90,35 @@ export async function retypeFile(file: File): Promise<File> {
 
 export async function extractArchivesRecursively(
   files: File[]
-): Promise<File[]> {
+): Promise<FileEntry[]> {
   const results = await Promise.all(
     files.map(async (file) => {
       if (file.type === 'zip') {
         return extractFilesFromZip(file);
       }
-      return [file];
+
+      const entry: FileEntry = { file, path: '' };
+
+      return [entry];
```

Comment:
Could just collapse this into `[{ file, path: '' }]`.

Reply:
Done